### PR TITLE
I 85 donor confirmation

### LINKED ISSUE #85: E03 S03 Doner confirmation.
Once Donors are notified, a Donor can accept the request. Upon accepting request there will be a record of donors request acceptance against the blood request post.

**Acceptance criteria:**
1. check if request is request post is valid
3. If the previously notified owner tries to accept an already accepted donor, they will see messages saying the request is complete.
4. Donor confirmation History.

### PR DESCRIPTION
**Pull request by Biplob14**
Fixes #85 
- Donor accepts blood donation request.
- Check if request still valid before accepting.
- Check if donor already accepted request.
- Store Donor acceptance request history.

- [x] Does not depend on other PR's
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Provide a brief overview of the changes introduced in this pull request.

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [x] I have followed the project's coding style and guidelines.
- [x] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [ ] I have checked for any potential conflicts with other branches or pull requests.
- [x] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: commons/dto/DonationDTO.ts [hunk 1]
```diff
@@ -33,3 +33,12 @@ export interface StepFunctionInput extends DTO {
...
   additionalParams?: Record<string, any>;
 }
+
+export type AcceptedDonationDTO = DTO & {
+  donorId: string;
+  requestPostId: string;
+  acceptanceTime: string;
+  seekerId: string;
+  createdAt: string;
+  status?: string;
```
Comment (sakibstark11) [inline commons/dto/DonationDTO.ts:43]: if its optional it should extend | concat the original type

File: iac/terraform/aws/donation/lambdas.tf [hunk 2]
```diff
@@ -19,6 +19,22 @@ locals {
...
+      invocation_arn_placeholder = "ACCEPT_DONATION_REQUEST_INVOCATION_ARN"
+      env_variables = {
+        DYNAMODB_TABLE_NAME = split("/", var.dynamodb_table_arn)[1]
+      }
     }
   }
 }
+
+
+# Error reading file "file:///home/sadekur/craftsmen/projects/BloodConnect/openapi/integration/aws/donations/:
```
Comment (sakibstark11) [inline iac/terraform/aws/donation/lambdas.tf:37]: ?

File: commons/dto/DonationDTO.ts (hunk 1 above)
Comment (Biplob14) [reply commons/dto/DonationDTO.ts:46]: Does [this](https://github.com/CraftsmenLtd/BloodConnect/pull/205/files#diff-a25adf68b1595e65a3ccaf232cd99cbfcd6cc47a5d4349ef1ee02c0a676652ff) change follows the structure you are talking about?

File: iac/terraform/aws/donation/lambdas.tf (hunk 2 above)
Comment (Biplob14) [reply iac/terraform/aws/donation/lambdas.tf]: Removed them the comment.

File: commons/dto/DonationDTO.ts (hunk 1 above)
Comment (sakibstark11) [reply commons/dto/DonationDTO.ts:46]: that link isn't taking me anywhere

**Review by sakibstark11 — APPROVED** [review]

File: commons/dto/DonationDTO.ts (hunk 1 above)
Comment (Biplob14) [reply commons/dto/DonationDTO.ts:46]: This link should have taken you to the code block shown you in following image. The change is done in this commit:
https://github.com/CraftsmenLtd/BloodConnect/pull/205/commits/34f6dba54869c4fac230dd4fa30a979ca6f674c2

image

**Review by sakibstark11 — APPROVED** [review]

**Pull request merged**